dotcom-rendering/src/web/components/Lightbox.tsx

```diff
@@ -0,0 +1,135 @@
+import { css } from '@emotion/react';
+import { space } from '@guardian/source-foundations';
+import { useEffect, useRef, useState } from 'react';
+import { LightboxPicture } from './LightboxPicture';
+import type { Props as LightboxProps } from './LightboxWrapper.importable';
+
+type Props = LightboxProps & {
+	isOpen: boolean;
+	onClose: () => void;
+	elements: LightboxProps[];
+};
+
+// Browsers apply some weird default styling to dialogs,
+// lets reset that!
+const dialogElementCssReset = css`
+	margin: 0;
+	max-width: none;
+	max-height: none;
+	box-sizing: border-box;
+`;
+
+export const Lightbox = ({
+	isOpen,
+	onClose,
+	elements,
+	...initialElementProps
+}: Props) => {
+	const dialogElement = useRef<HTMLDialogElement>(null);
+	const [index, setIndex] = useState(
+		elements.findIndex(
+			(element) => element.master === initialElementProps.master,
+		),
+	);
+
+	useEffect(() => {
+		/* eslint-disable @typescript-eslint/no-unsafe-call -- Typescript doesn't know these properties exist */
+		// @ts-expect-error -- Typescript's version of this interface is missing these properties
+		if (isOpen) dialogElement.current?.showModal();
+		// @ts-expect-error -- Typescript's version of this interface is missing these properties
+		else dialogElement.current?.close();
+		/* eslint-enable @typescript-eslint/no-unsafe-call */
+	}, [isOpen, dialogElement]);
+
+	const canPrev = () => index > 0;
+	const canNext = () => index < elements.length - 1;
+
+	const { caption } = elements[index];
+
+	return (
+		<dialog
+			ref={dialogElement}
+			css={css`
+				${dialogElementCssReset}
+
+				width: 100vw;
+				height: 100vh;
+				background-color: black;
+			`}
+		>
+			<div
+				css={css`
+					display: flex;
+					flex-direction: row;
+				`}
+			>
+				{/* We render all the images & show/hide the one we want to better support loading images
+					(& later, animations & swiping on mobile */}
+				{elements.map((element, elementIndex) => {
+					return (
+						<div
+							key={element.master}
+							id={`lightbox-image-${elementIndex}`}
+							css={css`
+								display: ${index === elementIndex
+									? 'block'
+									: 'none'};
+								flex-basis: 85%;
+							`}
+						>
+							<LightboxPicture
+								master={element.master}
+								alt={element.alt}
+							/>
+						</div>
+					);
+				})}
+				<div
+					css={css`
+						flex-basis: 15%;
+						align-items: flex-end;
+						display: flex;
+						flex-direction: column;
+						padding: ${space[2]}px;
+					`}
+				>
+					<div
+						css={css`
+							margin-bottom: ${space[2]}px;
+							color: white;
+						`}
+					>
+						<span>
+							{index + 1}/{elements.length}
+						</span>
+						<button
```

Comment:
Just a thought, is there benefit to using a source component here? I tend to lean towards using them even when it means a lot of style overrides but I'm open to questioning that approach. I know you mentioned there was still UI work needed here though so maybe this was always intended